backend/qemu.pm

```diff
@@ -164,7 +164,15 @@ sub _dbus_call {
 
 sub do_stop_vm {
     my $self = shift;
-    $self->{proc}->save_state();
+
+    my $proc = $self->{proc};
+    if ($bmwqemu::vars{QEMU_WAIT_FINISH}) {
+        # wait until QEMU finishes on its own; used in t/18-qemu-options.t
+        if (my $qemu_pid = $proc->qemu_pid) {
+            waitpid $qemu_pid, 0;
```

Comment:
Just a thought... how about using this in production, too, to avoid stale qemu processes? Assuming that not waiting can be a cause of that.

Reply:
That makes no sense. If the shutdown is not triggered within the SUT (which we can not generally assume¹) QEMU would continue to run forever and we would wait for it forever. We must signal QEMU to stop. That is done via the stop function provided by ReadWriteProcess which will use `waitpid` internally. Here it is only useful beause QEMU is started via `-version` in that unit test and therefore expected to terminate itself.

¹ The simplest case is when we forcefully end the test execution in the middle after an error occurred.